src/analysis/classic/druid/balance/modules/features/Abilities.ts

```diff
@@ -0,0 +1,94 @@
+import SPELLS from 'common/SPELLS/classic';
+import CoreAbilities from 'parser/core/modules/Abilities';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+
+class Abilities extends CoreAbilities {
```

Comment:
I've looked at multiple fights and added some more spells to Abilities.ts.

There are still some warnings _"Ability missing from spellbook"_, that fall in one of these categories:
* **Passive procs** (for example: Solar Eclipse procs when casting Starfire or Lunar Eclipse procs when casting Wrath). They are considered as a spell and the buff that appears afterwards. I've accounted for the buff though
* **Shapeshifting spells** (such as Cat Form or Moonkin Form)
* Spells in **other druid forms** (for example: switching to cat form then disappearing with Fade or speeding with Dash)
* **Healing spells** (such as Regrowth, Rejuvenation or Nourish)
* **Long lasting buffs** (such as Mark of the Wild or Gift of the Wild)

None of these spells are expected in a normal balance druid rotation during a fight. Some players may eventually use alternate forms during the fight (for example starting in hidden cat form, fleeing away in travel form or defensively switching to bear form to resist mobs), but this is very dependent of the group and the particular fight situation, so I do not believe it is of any interest to try to do something about them in the analyzer.

I see that the page is showing up properly, and I've see the same warnings for other classes. If I was to add all these other spells to Abilities.ts, this would clutter the interface with spells that are irrelevant for the balance spec, so all in all I think that the current state of this file is now consistent.

There are still a few cooldowns that need to be adjusted depending on the glyphs (for example https://www.wowhead.com/wotlk/item=45622 and https://www.wowhead.com/wotlk/item=43674), not sure how to handle them in Abilities.ts ?

Reply:
There are different [Spell Categories](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/dragonflight/src/parser/core/SPELL_CATEGORY.ts) for handling spells that are not part of the normal rotation. This puts the spells towards the bottom of the page so the person can focus on the spells specific to their spec but don't get penalized on downtime for casting them.

- **Passive procs** can be listed under `SPELL_CATEGORY.HIDDEN` to remove the console error. I imagine at some point a Balance Druid would love to see how "lucky" they got on procs.
- **Shapeshifting spells** - I'd use `SPELL_CATEGORY.UTILITY` 
- **Spells in Other Druid Forms** can be a judgement call between `SPELL_CATEGORY.UTILITY`  and `SPELL_CATEGORY.OTHERS`
- **Healing spells** should go under `SPELL_CATEGORY.OTHERS`. See [Retribution Paladin](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/dragonflight/src/analysis/classic/paladin/retribution/modules/features/Abilities.ts#L109) for an example. You can save time by copying spells from [Restoration Druid](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/dragonflight/src/analysis/classic/druid/restoration/modules/features/Abilities.ts) and updating the category.
- **Long lasting buffs** should be under `SPELL_CATEGORY.UTILITY`

For the glyphs, there would have to be a way to determine if someone is using them with the info provided by the log. I suggest `console.log`ing the `SELECTED_COMBATANT` or see if the spell has a different ID when the CD is shorter.